pkg/reconciler/v1alpha1/trigger/path/path.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package path
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/pkg/provisioners"
+)
+
+const (
+	prefix = "triggers"
+)
+
+// Generate generates the Path portion of a URI to send events to the given Trigger.
+func Generate(t *v1alpha1.Trigger) string {
+	return fmt.Sprintf("/%s/%s/%s", prefix, t.Namespace, t.Name)
+}
+
+// Parse parses the Path portion of a URI to determine which Trigger the request corresponds to. It
+// is expected to be in the form "/triggers/namespace/name".
+func Parse(path string) (provisioners.ChannelReference, error) {
+	parts := strings.Split(path, "/")
+	if len(parts) != 4 {
```

Comment:
If there is a "/" at the end then this may not work. How about strings.trim the first and last slashes and then split?

Reply:
My preference is to be strict in what we accept. If we find that this is a common error, then we can add the trim to help. I don't expect it to be, as for the most part it is just taking a string as given and using it as the URL.